diff --git a/dist/index.js b/dist/index.js
index 914285807221289f4a9b822fbcf18bec8e6da588..1470a5e4df4a68025de1a4b96444afb9432331ad 100644
--- a/dist/index.js
+++ b/dist/index.js
@@ -1,4 +1,5 @@
-import { getHighlighter } from 'shiki';
+// getHighlighter is deprecated in shiki v2+
+import { getSingletonHighlighter as getHighlighter } from 'shiki';
 import { visit } from 'unist-util-visit';
 import { toString } from 'hast-util-to-string';
 import rangeParser2 from 'parse-numeric-range';
diff --git a/package.json b/package.json
index 6d1e6d34025aa421ae4f4d6b302980ace4a60a54..ed83c94ba46e55ef21b9ba9eeae4131fbf1a9b86 100644
--- a/package.json
+++ b/package.json
@@ -38,7 +38,7 @@
     "mdast-util-to-hast": "^13.2.0",
     "prettier": "^3.3.2",
     "remark": "^15.0.1",
-    "shiki": "^1.9.1",
+    "shiki": "^2.4.1",
     "tsup": "^8.1.0",
     "tsx": "^4.15.7",
     "typescript": "^5.5.2",
